### PR TITLE
Add Schedule parameter to Publish-PerTenantExtensionApps

### PR DESCRIPTION
**Pull request by M4fin**
suggestion to add a schedule parameter to allow deployment of the app to the next major version via CI/CD Pipeline

### CODE REVIEW

**Comment by freddydk** [comment]
Could you add the documentation to the API description  on which the PR is based?

**Comment by M4fin** [comment]
Sure, documentation is [here](https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/administration/resources/dynamics_extensionupload#properties)

File: Saas/Publish-PerTenantExtensionApps.ps1 [hunk 1]
```diff
@@ -42,6 +42,8 @@ function Publish-PerTenantExtensionApps {
         $appFiles,
         [ValidateSet('Add','Force')]
         [string] $schemaSyncMode = 'Add',
+        [ValidateSet('Current','Next minor','Next major')]
+        [string] $schedule,
```
Comment (freddydk) [inline Saas/Publish-PerTenantExtensionApps.ps1:46]: Please modify the values to be the same as the supported values in schedule and then only set body.schedule if schedule is not blank.
Also we need to support not setting schedule at all - maybe allow blank and have default be blank.

**Review by freddydk — CHANGES_REQUESTED** [review]

File: Saas/Publish-PerTenantExtensionApps.ps1 (hunk 1 above)
Comment (M4fin) [reply Saas/Publish-PerTenantExtensionApps.ps1]: changed in the last commit

File: Saas/Publish-PerTenantExtensionApps.ps1 [hunk 2]
```diff
@@ -42,6 +42,8 @@ function Publish-PerTenantExtensionApps {
         $appFiles,
         [ValidateSet('Add','Force')]
         [string] $schemaSyncMode = 'Add',
+        [ValidateSet(' ','Current version','Next minor version','Next major version')]
```
Comment (freddydk) [inline Saas/Publish-PerTenantExtensionApps.ps1:45]: Please use empty string as default instead of a space.

**Review by freddydk — CHANGES_REQUESTED** [review]
And please add a line to the top section in releasenotes.txt, thanks

File: Saas/Publish-PerTenantExtensionApps.ps1 (hunk 2 above)
Comment (M4fin) [reply Saas/Publish-PerTenantExtensionApps.ps1:45]: Fixed

**Comment by M4fin** [comment]
@microsoft-github-policy-service agree

File: ReleaseNotes.txt [hunk 3]
```diff
@@ -16,6 +16,7 @@ If environment is specified as a Web Client URL, and BcAuthContext contains user
 Including caching of appinfos in CompilerFolder cache (to save time when caching on GitHub Actions)
 
 6.0.19
+Add Schedule parameter to Publish-PerTenantExtensionApps
```
Comment (freddydk) [inline ReleaseNotes.txt:19]: Please move the release note to 6.0.21

**Review by freddydk — CHANGES_REQUESTED** [review]

**Review by freddydk — APPROVED** [review]

File: Saas/Publish-PerTenantExtensionApps.ps1 [hunk 4]
```diff
@@ -42,6 +42,8 @@ function Publish-PerTenantExtensionApps {
         $appFiles,
         [ValidateSet('Add','Force')]
         [string] $schemaSyncMode = 'Add',
+        [ValidateSet('','Current version','Next minor version','Next major version')]
```
Comment (mazhelez) [inline Saas/Publish-PerTenantExtensionApps.ps1:45]: No need for `''`, it's not a mandatory parameter.

Comment (mazhelez) [inline Saas/Publish-PerTenantExtensionApps.ps1:45]:
```suggestion
        [ValidateSet('Current version','Next minor version','Next major version')]

```

File: Saas/Publish-PerTenantExtensionApps.ps1 [hunk 5]
```diff
@@ -42,6 +42,8 @@ function Publish-PerTenantExtensionApps {
         $appFiles,
         [ValidateSet('Add','Force')]
         [string] $schemaSyncMode = 'Add',
+        [ValidateSet('','Current version','Next minor version','Next major version')]
+        [string] $schedule = '',
```
Comment (mazhelez) [inline Saas/Publish-PerTenantExtensionApps.ps1:46]:
```suggestion
        [string] $schedule,

```

**Review by mazhelez — APPROVED** [review]

File: Saas/Publish-PerTenantExtensionApps.ps1 (hunk 4 above)
Comment (freddydk) [reply Saas/Publish-PerTenantExtensionApps.ps1:45]: most pipelines will have code where the value of this comes from a variable, which comes from a file or like. And this:

```
function test {
    Param(
        [ValidateSet('Current version','Next minor version','Next major version')]
        [string] $schedule
    )

    Write-Host "'$test'"
}

$schedule = ''
test -schedule $schedule
```

throws an error:
test : Cannot validate argument on parameter 'schedule'. The argument "" does not belong to the set "Current version,Next minor version,Next major version" specified by the ValidateSet attribute. Supply an 
argument that is in the set and then try the command again.

Which means that people will have to generate two code-paths for this scenario - and we do not want that.

**Pull request merged**